packages/project-config/tsconfig.json

```diff
@@ -1,6 +1,8 @@
 {
   "extends": "../../tsconfig.compilerOption.json",
   "compilerOptions": {
+    "moduleResolution": "NodeNext",
+    "module": "NodeNext",
```

Comment:
These or some variant of them seem necessary for type checking. It doesn't seem like they actually affect the emitted `*.d.ts` files. Still figuring that out.

There's a `bundler` option we could use as well for `moduleResolution` but I don't see the value in it yet